### PR TITLE
SEQNG-84: Dataset labels are not showing up in web client

### PR DESCRIPTION
**Pull request by cquiroz**
This PR does not fix the issue but identifies where changes are needed. This requires some updates on the engine side that are pending

### CODE REVIEW

File: modules/edu.gemini.seqexec.model/src/main/scala/edu/gemini/seqexec/model/Model.scala [hunk 1]
```diff
@@ -67,6 +67,7 @@ object Model {
     val breakpoint: Boolean
     val skip: Boolean
     val configStatus: Map[SystemName, ActionStatus]
+    val fileId: Option[dhs.ObsId]
```
Comment (cquiroz) [inline modules/edu.gemini.seqexec.model/src/main/scala/edu/gemini/seqexec/model/Model.scala]: I'm not sure this is the right place, Perhaps it should be part of the `observeStatus`. WDYT?

File: modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqexecEngine.scala [hunk 2]
```diff
@@ -103,6 +103,8 @@ class SeqexecEngine(settings: SeqexecEngine.Settings) {
             // TODO: Implement skipping at Engine level
             skip = false,
             configStatus = Map.empty,
+            // Where do I get the file id from?
+            None,
```
Comment (cquiroz) [inline modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqexecEngine.scala]: This is a key part of the engine update. We need to find how to populate this field from the DHS generated value

**Comment by jdnavarro** [comment]
I actually wanted to take care of this myself because this is also related to the task of creating distinct type for `Step[Action] \/ Step[Result]` and already have a local branch with some work done, but I misunderstood in a previous meeting that this was already taken care of. The idea we discussed at that time was to have something like:

```
sealed trait Step
case class StepPending(...)
case class StepDone(...)
```

And in the `StepDone` case class have the `fileId` parameter but not in the `StepPending` (or perhaps have it there too but wrapped in `Task`). This will get more intricate when we add more types of `Step`s we add but it's a good starting point for now.

Obtaining the actual `fileId` from the DHS should be part of an actual execution.

**Comment by cquiroz** [comment]
Sounds good 👍  I can do the frontend once your PR is merged. The change there is very minimal

**Pull request closed without merging**